joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -270,6 +293,66 @@ void Trajectory::interpolate_between_points(
                                 t[2] * 12.0 * coefficients[4] + t[3] * 20.0 * coefficients[5];
     }
   }
+  // Apply velocity, acceleration, and jerk limits with Ruckig
+  if ((period.seconds() > 0) && do_ruckig_smoothing)
+  {
+    ruckig::Ruckig<ruckig::DynamicDOFs> smoother(dim, period.seconds());
```

Comment:
It might seem strange to declare a new smoothing object with every cycle, but dim and period could change. So I think it has to be done this way.

https://github.com/pantor/ruckig/blob/master/include/ruckig/ruckig.hpp

I filed an issue asking if a method can be added to update in a nicer way.

Reply:
I would not say that `dim` can change, it is usually defined with the number of joint JTC is controlling. But yes, `update` in general may vary. Is there a possibility to update only the period?

Reply:
OK, I was confused about `dim` changing. I see that it can't now. I'll refactor this.